lib/inc/internal/ruby/ruby_value.hpp

```diff
@@ -79,12 +79,20 @@ namespace facter { namespace ruby {
          */
         leatherman::ruby::VALUE value() const;
 
+        /**
+         * Exposes an owned ruby VALUE as a facter ruby_value
+         * @param child the owned child object to wrap
+         * @return pointer to the ruby_value wrapper for the child object
+         */
+        ruby_value const* wrap_child(leatherman::ruby::VALUE child) const;
+
      private:
         static void to_json(leatherman::ruby::api const& ruby, leatherman::ruby::VALUE value, facts::json_allocator& allocator, facts::json_value& json);
         static void write(leatherman::ruby::api const& ruby, leatherman::ruby::VALUE value, std::ostream& os, bool quoted, unsigned int level);
         static void write(leatherman::ruby::api const& ruby, leatherman::ruby::VALUE value, YAML::Emitter& emitter);
 
         leatherman::ruby::VALUE _value;
+        mutable std::vector<std::unique_ptr<ruby_value>> children;
```

Comment:
Yeah, unfortunately the `lookup` contract assumes the value already exists and doesn't lend itself to creating one on demand.  Not sure how to solve this that doesn't require a larger refactoring.